hrflow/profile/matching.py

```diff
@@ -0,0 +1,113 @@
+import json
+
+from ..core.rate_limit import rate_limiter
+from ..core.validation import (
+    KEY_REGEX,
+    ORDER_BY_VALUES,
+    SORT_BY_VALUES,
+    validate_key,
+    validate_limit,
+    validate_page,
+    validate_provider_keys,
+    validate_reference,
+    validate_response,
+    validate_value,
+)
+
+
+class ProfileMatching:
+    def __init__(self, api):
+        """Initialize the ProfileMatching class with the provided API client."""
+        self.client = api
+
+    @rate_limiter
+    def list(
+        self,
+        source_key,
+        profile_key=None,
+        profile_reference=None,
+        source_keys=None,
+        page=1,
+        limit=30,
+        sort_by="created_at",
+        order_by=None,
+        created_at_min=None,
+        created_at_max=None,
+        **kwargs,
+    ):
+        """
+        💾 Match Profils indexed in Sources to a Profile
+        (https://api.hrflow.ai/v1/profils/matching).
+
+        Args:
+            source_key:         <string>
+                                The key of the Source in which the profile is indexed.
+            profile_key:        <string> (Optional)
+                                The key of a specific profile to macth with.
+            profile_reference:  <string> (Optional)
+                                The reference of a specific profile to macth with.
+            source_keys:        <list> (Optional)
+                                A list of keys for multiple Sources of profiles to be matched with the profile.
+            page:               <int> (default to 1)
+                                The page number for pagination.
+            limit:              <int> (default to 30)
+                                Number of profiles to fetch per page.
+            sort_by:            <string> (default to "created_at")
+                                The field to sort by.
+            order_by:           <string> (Optional)
+                                The order of sorting, either 'asc' or 'desc'.
+            created_at_min:     <string> (Optional)
+                                The minimum creation date of the profiles in format "YYYY-MM-DD".
+            created_at_max:     <string> (Optional)
+                                The maximum creation date of the profiles in format "YYYY-MM-DD".
+
+        Returns:
+            Match the profile identified by profile_key or profile_reference
+            and board_key with all profiles in the sources identified by keys in source_keys list.
```

Comment:
@Abdellahitech it's `source_key` instead of `board_key`

Reply:
corrigé